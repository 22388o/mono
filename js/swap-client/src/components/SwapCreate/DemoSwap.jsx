import React from 'react';
import { Button } from 'semantic-ui-react';

export const DemoSwap = ({ mockSwap }) => {
  return <>
    <p>Demo Swap</p>
    <Button circular secondary className='gradient-btn w-100 h-3' onClick={e => mockSwap({
      side: 'ask',
      baseAsset: 'BTC',
      baseNetwork: 'lightning.btc',
      baseQuantity: 0.001,
      quoteAsset: 'ETH',
      quoteNetwork: 'eth-l2.eth',
      quoteQuantity: 0.0000000000001})}>Swap 0.001 BTC for 0.0000000000001 ETH</Button>
    <Button circular secondary className='gradient-btn w-100 h-3' onClick={e => mockSwap({
      side: 'bid',
      baseAsset: 'ETH',
      baseNetwork: 'eth-l2.eth',
      baseQuantity: 0.0000000000001,
      quoteAsset: 'BTC',
      quoteNetwork: 'lightning.btc',
      quoteQuantity: 0.001})}>Swap 0.0000000000001 ETH for 0.001 BTC</Button>
  </>;
}