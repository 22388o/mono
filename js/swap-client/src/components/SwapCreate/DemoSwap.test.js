import { create, act } from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { DemoSwap } from './DemoSwap'

describe('DemoSwap test', () => {
  it('should work', () => {
    let tree
    let mockSwap = jest.fn();

    act(() => {
      tree = create(
          <DemoSwap
            mockSwap={mockSwap}
          />
      )
    })

    expect(tree).toMatchSnapshot()
  })
  afterAll(() => jest.resetModules())
})
