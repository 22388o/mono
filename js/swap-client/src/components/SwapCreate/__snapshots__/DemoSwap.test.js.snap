// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemoSwap test should work 1`] = `
[
  <p>
    Demo Swap
  </p>,
  <button
    circular="true"
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium gradient-btn w-100 h-3 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    secondary="true"
    tabIndex={0}
    type="button"
  >
    Swap 0.001 BTC for 0.0000000000001 ETH
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>,
  <button
    circular="true"
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium gradient-btn w-100 h-3 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    secondary="true"
    tabIndex={0}
    type="button"
  >
    Swap 0.0000000000001 ETH for 0.001 BTC
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>,
]
`;
